DigitalLearningSolutions.Data.Tests/Services/CourseServiceTests.cs

```diff
@@ -149,5 +149,21 @@ public void GetAllCoursesForDelegate_should_not_fetch_attempt_stats_if_course_no
             results[0].DelegateCourseInfo.Should().BeEquivalentTo(info);
             results[0].AttemptStats.Should().Be((0, 0));
         }
+
+        [Test]
+        public void VerifyUserCanAccessCourse_should_call_correct_data_service_method()
+        {
+            // Given
+            A.CallTo(() => courseDataService.GetCentreIdAndCategoryIdForCourse(A<int>._))
+                .Returns((2, 2));
+
+            // When
+            var result = courseService.VerifyUserCanAccessCourse(1, 2, 2);
+
+            // Then
+            A.CallTo(() => courseDataService.GetCentreIdAndCategoryIdForCourse(A<int>._))
+                .MustHaveHappened(1, Times.Exactly);
+            result.Should().Be(true);

```

Comment:
Do we want some tests to make sure the new method returns false when it should?

Reply:
Good idea, added two more to check if it returns false of the user has a centreId or categoryId that doesn't match the customisation